### PR TITLE
Remove call sites to Memory Reporter

### PR DESCRIPTION
**Pull request by kaidokert**
Memory Reporter was removed in https://github.com/youtube/cobalt/pull/925. This cleans up remaining
call sites and marks Starboard entry points to Memory Reporter fully deprecated.

b/291360030

### CODE REVIEW

**Comment by kaidokert** [comment]
Note internal change go/cobalt-cl/255600 needs to land first.

File: starboard/memory_reporter.h [hunk 1]
```diff
@@ -77,6 +79,9 @@ struct SbMemoryReporter {
 // Sets the MemoryReporter. Any previous memory reporter is unset. No lifetime
 // management is done internally on input pointer.
 //
+// NOTE: This module is unused starting with Starboard 15 and will be removed
+// in future.
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply starboard/memory_reporter.h]: Fixed, thanks

File: starboard/shared/starboard/memory.cc [hunk 2]
```diff
@@ -16,56 +16,31 @@
...
 }
 
 void* SbMemoryAllocate(size_t size) {
   void* memory = SbMemoryAllocateImpl(size);
-  SbReportAllocation(memory, size);
   return memory;
 }
 
 void* SbMemoryAllocateNoReport(size_t size) {
-  void* memory = SbMemoryAllocateImpl(size);
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply starboard/shared/starboard/memory.cc:62]: Please let me know if this doesn't seem right, resolving thread for now.

**Review by hlwarriner — APPROVED** [review]

File: starboard/memory_reporter.h [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 // Module Overview: Memory Reporting API.
 //
-// Provides an interface for memory reporting.
+// DEPRECATED: Provides an interface for memory reporting.
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply starboard/memory_reporter.h:17]: I intentionally didn't change the API in this CL at all. Starboard interface is fully unchanged here. The implementation is turned into a no-op though. If  we want to deprecate it in SB15, we can follow up with that, but technically we missed the window.

I'll close the thread for now - but lets separately follow up on that.

**Pull request merged**